.ci/README.md

```diff
@@ -1,63 +1,87 @@
 
-### `generate`
-`generate` takes two inputs:
-* The magic-modules repository, with the pull request's `head` checked out and the repo in "detached `HEAD`" state.
-* This CI repository.
+## What The Magician Does
+The Magician takes the PR you write against MagicModules and creates the downstream (generated) commits that are a result of your changes.  It pushes the downstream commits to the relevant downstream repositories, and opens PRs against them, referencing your open PR against MagicModules.  When those PRs are merged, it updates the submodules under `build/` and merges your MagicModules PR.
 
-It then runs `magic-modules/generate.yml`, which specifies one output:
+## Your Workflow
 
-* The magic-modules repository, after generation has been accomplished.
+You'll write some commits, likely modifying files under `provider` or `products`, and push a PR.  Since The Magician needs to be able to add commits to your branch to update submodules, **make sure the branch of your PR is in `GoogleCloudPlatform/magic-modules`**.
 
-After that, the generated repositories are uploaded to GitHub, in the concourse process runner's forks.  The pull request is updated to point to those forks.
+The Magician will take over, running the generator and running the downstream tests.  If the tests fail, you'll see "check failed" status on your PR<!--TODO(@ndmckinley) - better reporting and logs of test failures.-->.  When all the tests pass, the PR or PRs created by The Magician will link back to your MagicModules PR.  From there, it is your responsibility to engage with *all* the downstream PRs - The Magician doesn't test or merge those unless otherwise indicated.  You cannot merge your MagicModules PR until *all* downstreams accept the changes.  If a downstream requests changes and you need to modify your MagicModules PR, simply push another commit to your MagicModules PR and The Magician will push any generated updates to the downstream PRs.  Note that if you merge one PR, then another downstream asks for changes which affect the already-merged PR, you will have a messy situation - use caution when many repositories are impacted by your change.
 
-## Individual repo tests
+## Useful `fly` Invocations
 
-The second stage's goal is to confirm that the individual repos still pass tests.  It runs only after the first stage finishes.
+If you want to see what The Magician is going to do to your PR, you'll want to use the `fly` CLI.  You can find documentation about how to use `fly` [here](https://concourse-ci.org/fly.html), but the following invocations may be useful.
 
-### `terraform-test`
+First, you'll need to log in to https://terraform.ci.cloud-graphite.com and download `fly` from there.  You'll need to authenticate `fly` with that target, using `fly login`.  `fly login -t main` will prompt you interactively for the username and password.  [Here are the docs about fly login](https://concourse-ci.org/fly.html#fly-login).
```

Comment:
> log in to https://terraform.ci.cloud-graphite.com

Since this repo is public, maybe add that only Googlers have access to this.

Reply:
Also, should we at some point move it to something like magician.ci, or magic-modules.ci?

Reply:
We should!  I'll check on that.

Reply:
And I also want to add that I think we should look at alternate auth strategies - concourse has features where you can use github oauth to log in, for instance.